tests/test_vb_analytical.py

```diff
@@ -40,8 +40,31 @@ def given_noise(self, update_noise, check_method):
         variance = (sigma ** 2 * prior_sd ** 2) / denom
         scale = variance ** 0.5
 
+        # analytic solution based on
+        # https://www.econstor.eu/bitstream/10419/85883/1/02084.pdf eq 3 (whereever that is coming from)
+        exponent = (
+            -1
+            / 2
+            * (
+                np.dot(data, data) / sigma ** 2
+                + prior_mean ** 2 / prior_sd ** 2
+                - mean ** 2 / scale ** 2
+            )
+        )
+
+        z = (
+            (2 * np.pi) ** (-n / 2)
+            * sigma ** (-n)
+            * scale
+            / prior_sd
+            * np.exp(exponent)
+        )
+        logz = np.log(z)
+
         prior = bayes.vb.MVN(prior_mean, 1.0 / prior_sd ** 2)
-        gamma = bayes.vb.Gamma.FromMeanStd(1 / sigma ** 2, 42)
+
+        narrow_gamma_sd = 1.0e-4
+        gamma = bayes.vb.Gamma.FromMeanStd(1 / sigma ** 2, narrow_gamma_sd)
```

Comment:
Remaining issue: The analytic example just prescribes a fixed noise sigma and does not infer that (vb-option `update_noise=False`). So for a given `shape = X`, we compute the scale such that `shape * scale == 1/sigma**2`. As the update equations only contain `shape * scale` terms, they are independent from the initial choice of `X`. The free energy, however separates between shape and scale and `X` greatly influences the result. @aradermacher suggested to use a really narrow distribution, which I tried here.

Reply:
I would say that in case the noise is given, it is no longer part of the free energy equations (thus log(P(\Phi) should be removed, and I expect then that shape and scale should not longer have an influence.